src/annotator/host.coffee

```diff
@@ -4,6 +4,24 @@ Guest = require('./guest')
 
 module.exports = class Host extends Guest
   constructor: (element, options) ->
+
+    # Some options' values are not JSON-stringifiable (e.g. JavaScript
+    # functions) and will be omitted when the options are JSON-stringified.
+    # Add a JSON-stringifiable option for each of these so that the sidebar can
+    # at least know whether the callback functions were provided or not.
+    if options.services?[0]
+      service = options.services[0]
+      if service.onLoginRequest
+        service.onLoginRequestProvided = true
+      if service.onLogoutRequest
+        service.onLogoutRequestProvided = true
+      if service.onSignupRequest
+        service.onSignupRequestProvided = true
+      if service.onProfileRequest
+        service.onProfileRequestProvided = true
+      if service.onHelpRequest
+        service.onHelpRequestProvided = true
```

Comment:
This is one way for the annotator code to communicate to the sidebar code whether or not each of these callback functions has been provided. I'm open to other suggestions,